### ISSUE TITLE
IDE0051 false positive for extension referenced from another extension

### ISSUE THREAD

**Issue by MaceWindu** [dotnet/roslyn#82691]
**Version Used**: 

VS 18.4.0

**Steps to Reproduce**:

```cs
static class Helpers
{
    extension(User _this)
    {
        public bool IsUsed => _this.Helper();
    }

    // // warning IDE0051: Private member 'Helpers.Helper' is unused
    private static bool Helper(this User _this)
    {
        return _this != null;
    }
}

public class User
{
    public void Test()
    {
        _ = this.IsUsed;
    }
}

```

```xml
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<TargetFramework>net10.0</TargetFramework>
		<ImplicitUsings>enable</ImplicitUsings>
		<Nullable>enable</Nullable>

		<RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
		<AnalysisLevel>preview-All</AnalysisLevel>
		<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
	</PropertyGroup>
</Project>

```
**Diagnostic Id**:

[Remove unused private member (IDE0051)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0051)

**Expected Behavior**:

no reports

**Actual Behavior**:

incorrect diagnostic reported

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/roslyn/issues/81657
- https://github.com/dotnet/roslyn/issues/80645

**Comment by MaceWindu** [comment]
You are right - you are wrong

**Comment by igotinfected** [comment]
Seeing this also in our pipelines after `UseDotNet@2` picked up `10.0.200` over `10.0.103`:

```csharp
public static partial class OpenApiOptionsExtensions
{
    [GeneratedRegex(@"\/openapi\/v\d+(\.\d+)?\.json")]
    private static partial Regex SwaggerRegex();

    extension(OpenApiOptions options)
    {
        public void ConfigureXxxOpenApi(string title, string description)
        {
            options
                .ApplyDynamicServerUrl()
                ...;
        }

        public OpenApiOptions ApplyDynamicServerUrl()
        {
            options.AddDocumentTransformer(
                (document, context, _) =>
                {
                    ...
                    var subPath = httpContext.Request.Headers["X-Envoy-Original-Path"].FirstOrDefault() ?? string.Empty;
                    var servicePath = SwaggerRegex().Replace(subPath, string.Empty);
                    ...
```

> Found version 10.0.103 in channel 10.0 for user specified version spec: 10.0.x

> Found version 10.0.200 in channel 10.0 for user specified version spec: 10.0.x

**Comment by KUTlime** [comment]
Wonderfull. This breaks down my builds in CI/CD (_warnings as errors_). 🤦🏿‍♀️
